digitalearthau/coherence.py

```diff
@@ -139,18 +194,80 @@ def _check_ancestors(check_ancestors: bool,
                     s for s in siblings
                     if s != dataset and not s.is_archived and s.type == dataset.type
                 ]
-                if siblings:
+
+                # Do not process/check siblings for nbar/nbart/pq albers products
+                if siblings and (str(dataset.type.name) not in IGNORE_SIBLINGS):
                     _siblings_count += 1
                     sibling_ids = [str(d.id) for d in siblings]
-                    _LOG.info("dataset.siblings_exist",
+                    _LOG.info(str(dataset.type.name) + ".siblings.exists.dry_run",
                               dataset_id=str(dataset.id),
                               siblings=sibling_ids)
 
                     # Choose the most recent sibling and archive others
                     if archive_siblings:
-                        ancestors_archive_count += _archive_duplicate_siblings(dc, sibling_ids + [str(dataset.id)])
+                        _archive_count += _archive_duplicate_siblings(dc, sibling_ids + [str(dataset.id)])
+
+
+def _manage_downstream_ds(dc: Datacube,
+                          dataset: Dataset):
+    """
+    Need to manage the following two scenarios, with locationless source datasets:
+       1) Identify and list all the downstream datasets (may include level2, NBAR, PQ, Albers, WOfS, FC) that are either
+          Active or Archived.
+       2) Identify and list all the downstream datasets (includes summary products) that are either Active or Archived.
+    """
+
+    # Fetch all the derived datasets using the source dataset.
+    derived_dataset = _fetch_derived_datasets(dataset, dc)
```

Comment:
It looks like `_fetch_derived_datasets()` is being called twice (here and line 230), but I think it only needs to be called once. It already walks down the entire tree of child datasets.

Reply:
`_fetch_derived_datasets()` is first called with dataset to fetch all derived datasets. Then in the second call, the same function is called with locationless/archived dataset to fetch all the derived datasets of the same.